### PR TITLE
#131: Add network status and stats, as well as refactor out some time utilites

### PR DESCRIPTION
**Pull request by layneson**
- Refactored network library to distinguish between in and out feeds at runtime.
- Added sent/received byte counting to network library.
- Added an incoming feed status query/update system which includes sending heartbeats when a publishing feed is idle.
- Added a shared util header that currently contains some helpful time abstractions.
- Removed previous time code in r/bs and integrated new util stuff (much better).
- Updated network API calls in r/bs and added a call every loop which updates feed statuses.
- Added network throughput and rover feed status displays to the base station GUI.

### CODE REVIEW

**Comment by layneson** [comment]
[Basic network statistics / status](https://trello.com/c/ioNA2sXz/15-basic-network-statistics-status)

File: src/network/network.cpp [hunk 1]
```diff
@@ -201,64 +201,93 @@ Buffer get_outgoing_buffer(Feed* feed) {
...
     uint8_t ttl = MULTICAST_TTL;
 	setsockopt(sock_fd, IPPROTO_IP, IP_MULTICAST_TTL, &ttl, sizeof(ttl));
 
-    // Bind to the correct port.
-    struct sockaddr_in bind_addr{};
-    bind_addr.sin_family = AF_INET;
-    bind_addr.sin_port = htobe16(port);
-    bind_addr.sin_addr.s_addr = htobe32(INADDR_ANY);
+    switch (type) {
+    case FeedType::IN: {
```
Comment (AlexanderRBurke) [inline src/network/network.cpp:214]: Minor nit: I feel like the cases should be indented.

File: src/rover/main.cpp [hunk 2]
```diff
@@ -280,7 +275,7 @@ int main()
         }
 
 		// LIDAR stuff.
-		if (get_ticks() - last_lidar_send_time >= LIDAR_SEND_INTERVAL) {
+		if (lidar_send_timer.ready()) {
```
Comment (AlexanderRBurke) [inline src/rover/main.cpp:278]: This is definitely a good change

**Review by AlexanderRBurke — APPROVED** [review]
Good changes! I didn't have time to test this locally, but I trust you that it works.

**Pull request merged**